extensions/reactive-messaging-http/deployment/src/test/java/io/quarkus/reactivemessaging/http/sink/app/HttpEmitter.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.reactivemessaging.http.sink.app;
+
+import java.util.concurrent.CompletionStage;
+import java.util.function.Supplier;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+
+import io.vertx.core.buffer.Buffer;
+
+@ApplicationScoped
+public class HttpEmitter {
+
+    @Channel("my-http-sink")
+    Emitter<Object> emitter;
+
+    @Channel("http-sink-with-path-param")
+    Emitter<Object> emitterWithPathParam;
+
+    @Channel("retrying-http-sink")
+    Emitter<Object> retryingEmitter;
+
+    @Incoming("custom-http-source")
+    @Outgoing("custom-http-sink")
+    Message<String> passThroughWithCustomSerializer(Message<Buffer> message) {
```

Comment:
Any reason to not use plain payloads here?

String passThroughWithCustomSerializer(Buffer payload) {
   return payload.toString();
}

Reply:
no reason, it's the first time I played a bit more with reactive messaging, I was trying different things. Will switch to plain payloads for simplicity